### PR TITLE
Named tuple output for gsa

### LINKED ISSUE #697: Switch `gsa ` output to NamedTuple and add plotting convenience for them.
Currently we have the `SobolResult` and so on as the results from DiffEqSensitivity but these are all structures which do not clearly inform the user of the output's meaning. The output values should correspond to the input parameters and be a NamedTuple. We also need to have plot recipes(?) for the different outputs.

### PR DESCRIPTION
**Pull request by Vaibhavdixit02**
Fixes #697, to be followed up with `show` and plotting convenience from subsequent PRs.

### CODE REVIEW

File: src/estimation/gsa.jl [hunk 1]
```diff
@@ -3,23 +3,46 @@ function DiffEqSensitivity.gsa(m::PumasModel,subject::Subject,params::NamedTuple
     vhighparam = TransformVariables.inverse(toidentitytransform(m.param),p_range_high)
     p_range = [[vlowparam[i],vhighparam[i]] for i in 1:length(vlowparam)]
     trf_ident = toidentitytransform(m.param)
+    sim_ = simobs(m,subject,params,args...;kwargs...)
+    length_vars = append!([0],cumsum([length(getproperty(sim_.observed,key)) for key in vars]))
```
Comment (andreasnoack) [inline src/estimation/gsa.jl]:
```suggestion
    length_vars = append!([0], cumsum([map(length, sim_.observed)...]))
```
Generally, you should avoid using `getproperty` when accessing `NamedTuple`s. Also, please add spaces after commas and around most operators to make the code more readable.

File: src/estimation/gsa.jl [hunk 2]
```diff
@@ -3,23 +3,46 @@ function DiffEqSensitivity.gsa(m::PumasModel,subject::Subject,params::NamedTuple
...
+    params_len = append!([0],cumsum([length(param) for param in params]))
```
Comment (andreasnoack) [inline src/estimation/gsa.jl]:
```suggestion
    params_len = append!([0], cumsum([map(length, params)...]))
```

File: src/estimation/gsa.jl [hunk 3]
```diff
@@ -3,23 +3,46 @@ function DiffEqSensitivity.gsa(m::PumasModel,subject::Subject,params::NamedTuple
...
     function f(p)
         param = TransformVariables.transform(trf_ident,p)
         sim = simobs(m,subject,param,args...;kwargs...)
         collect(Iterators.flatten([getproperty(sim.observed,key) for key in vars]))
```
Comment (andreasnoack) [inline src/estimation/gsa.jl]:
```suggestion
        collect(Iterators.flatten([sim.observed[key] for key in vars])) 
```

File: src/estimation/gsa.jl [hunk 4]
```diff
@@ -3,23 +3,46 @@ function DiffEqSensitivity.gsa(m::PumasModel,subject::Subject,params::NamedTuple
...
+    return sens_result(sensitivity,params,vars,length_vars,params_len,trf_ident)
 end
 
 function DiffEqSensitivity.gsa(m::PumasModel,population::Population,params::NamedTuple,method::DiffEqSensitivity.GSAMethod,vars = [:dv],p_range_low=NamedTuple{keys(params)}([par.*0.05 for par in values(params)]),p_range_high=NamedTuple{keys(params)}([par.*1.95 for par in values(params)]),args...; kwargs...)
     vlowparam = TransformVariables.inverse(toidentitytransform(m.param),p_range_low)
     vhighparam = TransformVariables.inverse(toidentitytransform(m.param),p_range_high)
     p_range = [[vlowparam[i],vhighparam[i]] for i in 1:length(vlowparam)]
     trf_ident = toidentitytransform(m.param)
+    sim_ = simobs(m,population,params,args...;kwargs...)
+    length_vars = append!([0],cumsum([length(getproperty(sim_.sims[1].observed,key)) for key in vars]))
```
Comment (andreasnoack) [inline src/estimation/gsa.jl]:
```suggestion
    length_vars = append!([0], cumsum([map(length, sim_.sims[1].observed)...]))
```
What if the length of the vectors in `observed` differ between the subjects?

File: src/estimation/gsa.jl [hunk 5]
```diff
@@ -3,23 +3,46 @@ function DiffEqSensitivity.gsa(m::PumasModel,subject::Subject,params::NamedTuple
...
         param = TransformVariables.transform(trf_ident,p)
         sim = simobs(m,subject,param,args...;kwargs...)
         collect(Iterators.flatten([getproperty(sim.observed,key) for key in vars])) 
     end
-    DiffEqSensitivity.gsa(f,p_range,method)
+    sensitivity = DiffEqSensitivity.gsa(f,p_range,method)
+    return sens_result(sensitivity,params,vars,length_vars,params_len,trf_ident)
 end
 
 function DiffEqSensitivity.gsa(m::PumasModel,population::Population,params::NamedTuple,method::DiffEqSensitivity.GSAMethod,vars = [:dv],p_range_low=NamedTuple{keys(params)}([par.*0.05 for par in values(params)]),p_range_high=NamedTuple{keys(params)}([par.*1.95 for par in values(params)]),args...; kwargs...)
```
Comment (andreasnoack) [inline src/estimation/gsa.jl]:
```suggestion
function DiffEqSensitivity.gsa(
  m::PumasModel,
  population::Population,
  params::NamedTuple,
  method::DiffEqSensitivity.GSAMethod,
  vars = [:dv],
  p_range_low=NamedTuple{keys(params)}([par.*0.05 for par in values(params)]),
  p_range_high=NamedTuple{keys(params)}([par.*1.95 for par in values(params)]),
  args...; kwargs...)
```
What's `0.05` and `1.95`?

File: src/estimation/gsa.jl [hunk 6]
```diff
@@ -3,23 +3,46 @@ function DiffEqSensitivity.gsa(m::PumasModel,subject::Subject,params::NamedTuple
...
     trf_ident = toidentitytransform(m.param)
+    sim_ = simobs(m,population,params,args...;kwargs...)
+    length_vars = append!([0],cumsum([length(getproperty(sim_.sims[1].observed,key)) for key in vars]))
+    params_len = append!([0],cumsum([length(param) for param in params]))
     function f(p)
         param = TransformVariables.transform(trf_ident,p)
-        sims = simobs(m,population,param,args...;kwargs...)
-        collect(Iterators.flatten([collect(Iterators.flatten([getproperty(sims[i].observed,key) for key in vars])) for i in 1:length(sims)]))
+        sim = simobs(m,population,param,args...;kwargs...)
+        mean([collect(Iterators.flatten([getproperty(sim.sims[i].observed,key) for key in vars])) for i in 1:length(sim.sims)])
```
Comment (andreasnoack) [inline src/estimation/gsa.jl]:
```suggestion
        mean(collect(Iterators.flatten(_sim.observed[key] for key in vars)) for _sim in sim.sims)
```
Would it make sense to have a version of `gsa` that worked for `NamedTuple`s?

Comment (Vaibhavdixit02) [reply src/estimation/gsa.jl]: I am not sure, having a vector interface in DiffEqSensitivity allows more flexibility, I think. Especially as it fits well with the rest of the DiffEq ecosystem.

**Comment by Vaibhavdixit02** [comment]
Just to clarify, the show method that I have added right now is not very elegant and I will revisit it later. I have done so because of the immediate need for the work on the ACoP poster. (@vjd) 

The current output looks like this. 

```julia> sobol = gsa(m_diffeq,
                   ev2,
                   p,
                   DiffEqSensitivity.Sobol(N=10,order=[0,1]),
                   [:auc])
First Order Indices

Derived Variable: auc

Parameter first order indices 

θ          [0.0, 0.44223, -0.000489486]

Total Order Indices

Derived Variable: auc

Parameter total order indices 

θ          [0.0, 0.359434, 6.26661e-6]
```

**Comment by ChrisRackauckas** [comment]
Works for now.

**Comment by vjd** [comment]
yeah. it's ok for now. We can create a lookup dictionary for the parameter names when dealing with the pbpk model

**Pull request merged**